streaming/base/dataset.py

```diff
@@ -356,33 +321,35 @@ def __init__(self,
         # Note: we do not assume that the end of __iter__() will ever be reached, so we need to
         # increment the epoch counter at the start of __iter__() instead of at the end, so we need
         # to track what the next epoch is, not the current epoch.
-        next_epoch_shm = CreateSharedMemory(name=f'{self._prefix}_next_epoch',
+        next_epoch_shm = CreateSharedMemory(name=f'{self._shm_prefix}_next_epoch',
                                             size=np.int64().nbytes)
         self._next_epoch_shm = next_epoch_shm.shm
         self._next_epoch_arr = np.ndarray(1, buffer=self._next_epoch_shm.buf, dtype=np.int64)
         self._next_epoch_arr[0] = 0
 
         # Get the filelock filename that protects shard_states shared memory array.
-        self.shard_states_filename = os.path.join(self._shared_dir, '_shard_states_filelock')
+        self.shard_states_filename = os.path.join(self._filelock_root, '_shard_states_filelock')
 
         # Create or attach shard_states array (tells if each shard is unknown, downloading, or
         # downloaded).
-        shard_states_shm = CreateSharedMemory(name=f'{self._prefix}_shard_states',
+        shard_states_shm = CreateSharedMemory(name=f'{self._shm_prefix}_shard_states',
                                               size=self.num_shards * np.uint8(0).nbytes)
         self._shard_states = shard_states_shm.shm
 
-        # Destroy process group, and de-initialize the distributed package
-        barrier()
-        if is_dist_pg_initialized:
-            dist.destroy_process_group()
-
         # Placeholder for a shared memory object where load_state_dict() saves its data to be
         # picked up by __iter__().
         self._resume_shm = None
 
         # Placeholder for a _PartitionState which tracks state during __iter__().
         self._partition_state = None
 
+    def __del__(self) -> None:
+        """Destructor, which releases its local working directories."""
+        try:
+            self._locals_shm.buf[:4] = np.int32(0).tobytes()
```

Comment:
you need to check if `_local_shm` is present or not, e.g. if hasattr(self, '_locals_shm') because if the streaming dataset crash before allocating `self._locals_shm` then `__del__` will fail with noattributeerror.

Reply:
done